app/components/markdown/markdown.js

```diff
@@ -446,7 +501,14 @@ export default class Markdown extends PureComponent {
     };
 
     render() {
-        let ast = this.parser.parse(this.props.value.toString());
+        let ast;
+
+        if (this.props.enableLatex && this.props.enableInlineLatex) {
+            const escapedContent = escapeLatexInputPreParser(this.props.value.toString());
```

Comment:
The way I solved this in the web app PR was by modifying the markdown parser. I made the parser stop at `$` and then check if there was any latex code ahead. If there was latex code ahead it was parsed into the AST as one big block and the different markdown syntax components where ignored (like `_emphasis_`).

I tried to fix this problem by adding a transformation as you described but this didn't work when there were markdown syntax compontents inside of the inline latex code. For example: `$x_0 = x_1 = /cdots$` would not render correctely because the markdown library would parse `0 = x` as a block that would needed to get emphasis. 

It is possible to overcome this issue by manipulating the AST and checking if there is special syntax inside inline latex code as you said. But that seemed a lot more complicated than doing it this way.

Reply:
Perhaps it would be easier to just modify the Markdown library to do that then. I think I'd feel more comfortable doing that than escaping and unescaping these since the current way of doing this feels like it could be fragile since anything else handled to the Markdown code could interfere with this as it currently is. For example, if at-signs or number signs are used in the Latex, I think that'll cause the Markdown parser to interfere since the mobile app's version is actually responsible for doing at-mentions and hashtags as well.

Overall, I don't think it should be that hard to add since [our fork of commonmark.js](https://github.com/mattermost/commonmark.js) is much better maintained, and I've generally found it easier to work with than marked, but it does require changes to two libraries to change: that fork and https://github.com/mattermost/commonmark-react-renderer.

You'll have to make most of your changes in [`inlines.js`](https://github.com/mattermost/commonmark.js/blob/master/lib/inlines.js) by adding an extra case to `parseInlines`. It's a bit more powerful than the `marked` version, so you can backtrack to check what's before the $ without having to use a lookbehind (see the at mention or hashtag parsing for an example of that).

Then, you'll want to add a new `Node` type for the inline latex and ensure commonmark-react-renderer has some basic handling for it since it needs to know about every potential type of element for some reason that I can't quite remember.

Apologies for making this more complicated, but I think avoiding the escaping logic will be safer in the long run.

Reply:
I understand that this way of doing things is indeed a bit safer and more robust so I'll take a look at this in the following week.

I thought about changing the commonmark library when I started working on this PR but I looked at the packages and it seemed like the app used the standard commonmark packages. Then I looked into the patches that are done after the npm install process and I guess that is where the changes occur.

So how do I change the commonmark library's? Do I just make a PR on the mattermost fork and you guys will change the patch file or how does this work?